float64 singleX
float64 singleY

---

float64 valx
float64 valy